tally_ho/apps/tally/models/result_form.py

```diff
@@ -252,6 +252,14 @@ def audit_recommendation(self):
             recomendation_index = self.audit.resolution_recommendation
             return ClearanceResolution.choices()[recomendation_index.value][1]
 
+    @property
+    def audit_quaritine_check_name(self):
+        if self.audit:
+            quarantine_checks = self.audit.quarantine_checks.all()
```

Comment:
can't you call `first` here to accomplish the result below?

but, can't the different quarantine checks connected to an audit have different names?

Reply:
You have a point, i just remembered it's many to many relationship, so i will just have to them all, right?. And yes `first` would have done the job since it returns `None` when the `queryset` is empty.

Reply:
I think this is a one-many, each audit can link to many quarantine checks.

Cool, so I think this brings up 2 questions to answer before we can continue:

1) Does it make sense for an audit to link to multiple quarantine checks?
2) If above is yes, does it make sense to show the names of any unique quarantine checks?

I think the answer to (2) is yes, but it depends on the answer to (1). If (1) is false we can update the model to have this be a one-one relationship, if the answer to (1) is true, we can do an all or an all by unique name call and string together all the names somehow

Reply:
@akahure do you know the answer to (1) above?

Reply:
Cool, so currently on the code it's a many-to-many relationship, but i think it should be a one-to-many instead because one audit can assigned more than one quarantine check when a result form fails more then one quarantine check.

Reply:
On second thought i think the way it is okay, because you might have one `quarantine check` assigned to several `audits` and vice versa, so we need to display all `quarantine checks` for a particular `audit`.

Reply:
Why do we need multiple `quarantine checks` attached to an audit?

Reply:
when a result form fails more then one quarantine check. have a look at this, https://github.com/onaio/tally-ho/blob/495b73171a1fa524b02ece7034d42ab04c6956ee/tally_ho/apps/tally/views/quality_control.py#L185 and https://github.com/onaio/tally-ho/blob/495b73171a1fa524b02ece7034d42ab04c6956ee/tally_ho/libs/verify/quarantine_checks.py#L107

Reply:
ah ok makes sense, so do you want to rewrite this to get any uniquely named quarantine checks and the below to show all their names?

Reply:
All `quarantine checks` are uniquely named from what i can see here: https://github.com/onaio/tally-ho/blob/495b73171a1fa524b02ece7034d42ab04c6956ee/tally_ho/libs/verify/quarantine_checks.py#L7 I think i should just refactor that logic to pull the constants e.g. quarantine checks from settings, and finally i'll just show all names of the quarantine checks assigned to an audit since they are unique.

Reply:
Good point so you could ensure unique on ID which is better, what I was thinking was if you can assign the same quarantine check to a single audit, and that's what we'd want to be unique, but if that's not possible (which [the docs](https://docs.djangoproject.com/en/3.0/topics/db/examples/many_to_many/) make at least sound non-standard) then we don't need to worry about that

Reply:
Cool, by `if you can assign the same quarantine check to a single audit` do you mean like assigning the same quarantine check more than once to a single audit? If yes then that can not happen in our case because we assign quarantine checks one at a time to an audit by looping through the unique checks. If no, then please explain :)

Reply:
Yep, that's exactly what I mean. Cool, that makes sense, and once an audit has been created we can't go back to it and add additional checks.

I'm going to mark this conversation as resolved and add a note on the lines below about handling all of the results from this query.